components/motor/ulnstepper/28byj-48.go

```diff
@@ -372,12 +377,15 @@ func (m *uln28byj) SetRPM(ctx context.Context, rpm float64, extra map[string]int
 
 // Set the current position (+/- offset) to be the new zero (home) position.
 func (m *uln28byj) ResetZeroPosition(ctx context.Context, offset float64, extra map[string]interface{}) error {
+	newPosition := int64(-1 * offset * float64(m.ticksPerRotation))
 	if err := m.Stop(ctx, extra); err != nil {
 		return err
 	}
 	m.lock.Lock()
 	defer m.lock.Unlock()
-	m.stepPosition = int64(-1 * offset * float64(m.ticksPerRotation))
+	m.stepPosition = newPosition
+	m.targetStepPosition = newPosition
+	// use Stop to set the target position to the current position again
```

Comment:
so should we call stop here then?

Reply:
no, I should move this comment above the stop, if we stop when we're already in lock we deadlock the routine and hang.